packages/babel-parser/test/fixtures/typescript/types/union-intersection/input.ts

```diff
@@ -7,3 +7,4 @@ type J = number | string
 type F = number & string
 type K = | number | string
 type M = & number & string
+type N = | number
```

Comment:
can you add test for `& number`

https://github.com/typescript-eslint/typescript-eslint/blob/teest/estree-union-intersection/packages/shared-fixtures/fixtures/typescript/basics/union-intersection.src.ts

Reply:
Yes, but is actually the same result.

Reply:
its more about to ensre that no one breaks it in feature, in case of refactor